variadics/src/lib.rs

```diff
@@ -55,6 +55,24 @@ macro_rules! var_args {
     ($a:pat, $( $b:tt )+) => ( ($a, $crate::var_args!( $( $b )* )) );
 }
 
+/// return the (top-level) length of en explicit variadic
+/// (cannot pass in a nested macro call like var_expr!(...))
+#[macro_export]
+macro_rules! var_len {
+    // Match the empty tuple or base case
+    (()) => {
+        0
+    };
+    // Match a single element tuple, terminating the recursion
+    (($elem:expr,())) => {
+        1
+    };
+    // Match deeper nested tuples
+    (($head:expr, $tail:tt)) => {
+        1 + var_len!($tail)
+    };
+}
+
```

Comment:
This doesn't seem to be used for anything meaningful?

Reply:
True -- not that handy as it only works with explicit arguments being passed in. I will delete.